src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -161,20 +161,21 @@ internal CommitManager(ImmutableArray<char> potentialCommitCharacters, RecentIte
 
                 edit.Replace(mappedSpan.Span, change.TextChange.NewText);
 
-                // The edit updates the snapshot however other extensions may make changes there.
-                // Therefore, it is safe to consume subjectBuffer.CurrentSnapshot for further calculations.
-                edit.Apply();
+                var updatedCurrentSnapsot = edit.Apply();
                 
                 if (change.NewPosition.HasValue)
-                {
-                    view.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(subjectBuffer.CurrentSnapshot, change.NewPosition.Value));
+                { 
+                    // Use the most recent snapshot to make a snapshot point.
+                    view.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(updatedCurrentSnapsot, change.NewPosition.Value));
                 }
 
                 includesCommitCharacter = change.IncludesCommitCharacter;
 
                 if (roslynItem.Rules.FormatOnCommit)
                 {
                     // refresh the document
+                    // The edit updates the snapshot however other extensions may make changes there.
+                    // Therefore, it is safe to consume subjectBuffer.CurrentSnapshot for further calculations rather than the updated current snapsot defined above.
                     document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
❓ Is there any reason to *not* use `updatedCurrentSnapshot` here? It seems like that would be the easier approach to explain, but if it's not possible then I'd like to see the preceding comment expanded to state why `subjectBuffer.CurrentSnapshot` is *needed* as opposed to the current comment that only mentions why it is *allowed*.

Reply:
@sharwell The editor supports extensions creating further text buffer edits after an edit is Applied. There's actually an entire design for this, for example [special events](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.itextbuffer.postchanged) that are raised precisely to understand this situation. If you don't fetch with the latest snapshot here, you're operating on an outdated snapshot which means GetFormattingChangesAsync would be operating on a stale buffer, potentially mangling the document.

Reply:
Whereas, the TryMoveCaretToAndEnsureVisible needs to use updatedCurrentSnapshot: change.NewPosition.Value is a point relative to that precise snapshot, and the constructor could potentially throw if you used CurrentSnapshot and any intervening edits deleted spans.

Reply:
Thanks for the update. It sounds like this is the confusing part:

> Therefore, it is **safe to consume** subjectBuffer.CurrentSnapshot

This should be updated to reflect that it is _required_ to use `subjectBuffer.CurrentSnapshot` because if any changes were made during the event handlers triggered by applying the previous edit, those changes will not be included in `updatedCurrentSnapshot`.